lib/functions/RegularFunctions.ts

```diff
@@ -94,6 +94,10 @@ const equality = {
   overloads: declare(C.RegularOperator.EQUAL)
     .numberTest(() => (left, right) => left === right)
     .stringTest(() => (left, right) => left.localeCompare(right) === 0)
+    .set(
+      [ TypeURL.RDF_LANG_STRING, TypeURL.RDF_LANG_STRING ],
+      () => ([ left, right ]) => bool(RDFTermEqual(left, right)),
```

Comment:
When using the rdf term equal it would also check if the datatype is the same. Our type system is able to handle subtypes. I think it would be better to support that feature. (This just means you need to check value and language by yourself instead of using RDFTermEqual)

Reply:
Oh, I see you where commenting on my previous, sorry :sweat_smile:

Reply:
okay, will do

Reply:
Is the error needed when two TypeURL.RDF_LANG_STRINGs are not equal?

If that is the case it wonder if RDFTermEqual might be fine.

Reply:
I don't know if I understand your question. So we will implement a languageString equality so that it does not throw an error when they are not equal. Instead, we want it to return false. 

We can not use RDFTermEqual since it will check if the datatype is also equal. However, Sparqlee can discover subTypes. In other words, if I define `myLangStr`, a subtype of `languageString`, it should still work i the equality. (As in, this is how all other equality functions also work)

Reply:
> So we will implement a languageString equality so that it does not throw an error when they are not equal. Instead, we want it to return false.

Ah okay, yes that would be an answer to the question. So I also have change the test that changed before.
I am on it.